addons/main/functions/fnc_showDownedSkull.sqf

```diff
@@ -55,37 +55,43 @@ if (_show) then {
         GVAR(downedBlur) ppEffectAdjust [0];
         GVAR(downedBlur) ppEffectCommit 0;
     };
-    GVAR(skullID) = _time spawn {
-        private _waitTime = _this / 2;
-        private _maxTimePerPart = _waitTime / 7;
-        private _controls = + (uiNamespace getVariable [QGVAR(skullControls), []]);
-        (_controls select 0) ctrlSetFade 0;
-        (_controls select 0) ctrlCommit (random [_this / 4, _waitTime, _this]);
-        _controls deleteAt 0;
-        _controls = _controls call BIS_fnc_arrayShuffle;
-        sleep _waitTime;
+
+    GVAR(skullActive) = true;
+    private _controlsCopy = + (uiNamespace getVariable [QGVAR(skullControls), []]);
+    _controlsCopy = _controlsCopy call BIS_fnc_arrayShuffle;
+    private _waitTime = _time / 2;
+    private _maxTimePerPart = _waitTime / 7;
+    private _first = _controlsCopy deleteAt 0;
+    _first setVariable [QGVAR(timeToShow), random [_time / 4, _waitTime, _time]];
+    {
+        _x setVariable [QGVAR(timeToShow), random [_waitTime / 4, _waitTime / 2, _waitTime]];
+        private _rnd = (random _maxTimePerPart) min _waitTime;
+        _waitTime = (_waitTime - _rnd) max 0;
+    } forEach _controlsCopy;
+    _controlsCopy pushBack _first;
+    [{
+        params ["_controls", "_unit"];
+        private _timeLeft = (_unit getVariable [QGVAR(bleedoutKillTime), -1]) - cba_missionTime;
         {
-            _x ctrlSetFade 0;
-            _x ctrlCommit (random [_waitTime / 4, _waitTime / 2, _waitTime]);
-            private _rnd = (random _maxTimePerPart) min _waitTime;
-            _waitTime = (_waitTime - _rnd) max 0;
-            sleep _rnd;
+            _x ctrlSetFade (linearConversion [_x getVariable [QGVAR(timeToShow), 0], 0, _timeLeft, 1, 0, true]);
```

Comment:
Skull seems a little sluggish coming in compared to before, could set the maxFrom to 4 or so.

Reply:
The time to appear needs to be tweaked. Right now it only checks when to start showing the skill but it is fully opaque when the timer hits zero.

Before it could be opaque way before. Which imo is a better solution so I need to tweak the linearConversion a bit.